### PR TITLE
S2U-25 5.1.1.21 Test & Quizzes: Question pools export - MASTER

### PR DESCRIPTION
**Pull request by jesusmmp**
https://sakaiproject.atlassian.net/browse/S2U-25

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java [hunk 1]
```diff
@@ -461,59 +463,102 @@ public void setAuthorizationBean(AuthorizationBean authorizationBean)
...
-  public void importPoolFromQti(ValueChangeEvent e)
-  {
+  public void importPool(ValueChangeEvent e) {
+    String sourceType = ContextUtil.lookupParam("sourceType");
     String uploadFile = (String) e.getNewValue();
 
-    try
-    {
-    	processAsPoolFile(uploadFile);
+    if (uploadFile!= null && uploadFile.startsWith("SizeTooBig:")) {
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java:470]: use StringUtils

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java [hunk 2]
```diff
@@ -461,59 +463,102 @@ public void setAuthorizationBean(AuthorizationBean authorizationBean)
...
+        Long sizeMax = Long.valueOf(ServerConfigurationService.getString("samigo.sizeMax", "20480"));
+        String sizeTooBigMessage = MessageFormat.format(rb.getString("import_size_too_big"), uploadFile.substring(11), Math.round(sizeMax.floatValue()/1024));
+        FacesMessage message = new FacesMessage(sizeTooBigMessage);
+        FacesContext.getCurrentInstance().addMessage(null, message);
+        // remove unsuccessful file
+        log.debug("****Clean up file: {}", uploadFile);
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java:476]: maybe only output the first so many chars

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java [hunk 3]
```diff
@@ -461,59 +463,102 @@ public void setAuthorizationBean(AuthorizationBean authorizationBean)
...
-	log.error(e.getMessage(), e);
+    finally {
+        boolean success = false;
+        // remove unsuccessful file
+        if (!fileNotFound) {
+          log.debug("****Clean up file: {}", fileName);
+          File f1 = new File(fileName);
+          success = f1.delete();
+          if (!success) {
+            log.error("Failed to delete file {}", fileName);
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java:537]: probably should be a warn

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/cp/DownloadCPServlet.java [hunk 4]
```diff
@@ -79,113 +88,203 @@ public void doGet(HttpServletRequest req, HttpServletResponse res)
...
+				Map.Entry entry = (Map.Entry) it.next();
+				filename = (String)  entry.getKey();
+				ze = new ZipEntry(filename.substring(1));
+				zos.putNextEntry(ze);
+				b = (byte[]) entry.getValue();
+				zos.write(b, 0, b.length);
+				zos.closeEntry();
+			}
+		} catch (IOException e) {
+			log.error(e.getMessage());
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/cp/DownloadCPServlet.java:277]:
```suggestion
			log.error(e.toString());
```

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java (hunk 2 above)
Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java]: Solved

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java (hunk 3 above)
Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java]: Changed!

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/cp/DownloadCPServlet.java (hunk 4 above)
Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/cp/DownloadCPServlet.java]: Changed!

**Comment by mpellicer** [comment]
Thanks @jesusmmp , did you adress all the community comments?

**Comment by jesusmmp** [comment]
> Thanks @jesusmmp , did you address all the community comments?

Yes, all suggestions were changed.

**Review by mpellicer — APPROVED** [review]

**Comment by mpellicer** [comment]
Thank you very much Jesús @jesusmmp !

**Pull request merged**